ldk/javascript/src/webpack/check-loop-open-handler-plugin.ts

```diff
@@ -0,0 +1,29 @@
+import { Compiler, WebpackError } from 'webpack';
+
+const assetName = 'loop.js';
+const warningMsg =
+  'Please provide a way for users to open your Loop. You can do this using the ui.loopOpenHandler. This is required to submit your Loop.';
+const plugin = 'CheckLoopOpenHandlerPlugin';
+export class CheckLoopOpenHandlerPlugin {
+  apply(compiler: Compiler): void {
+    compiler.hooks.thisCompilation.tap(plugin, (compilation) => {
+      compilation.hooks.processAssets.tap({ name: plugin }, (assets) => {
+        const loopjs = assets[assetName];
+        if (loopjs) {
+          const fileContent = loopjs.source();
+          if (
```

Comment:
I'm sure this is better than what we have now, I just want to point out some scenarios that this might handle correctly:

- The loop.js file is not minimized
- Some optimization that renames `ui` to something else
- Console log messages that have this string in them
- Commented out code that has an uncommented version that *does* make this call

Reply:
@iangallOlive those are all good/valid scenarios. I think the last statement is the most likely as where we intercept the loop js file at is not in its fully optimized form.

I tried to find a better solution but was unable to unless we wanted to read over each file in a project (didnt think that was great either)

I can try to update the logic to handle that last scenario more smoothly

Reply:
I can change to use this `compilation.hooks.afterOptimizeAssets` this hook will give us access to the optimized loop js file which does not include any comments and we could just loop for `loopOpenHandler(`

Reply:
Yeah, and I wasn't trying to criticize your work, just that it's a really hard problem to fully solve.

I think checking after optimization is a great idea, and might solve everything. My only question is: what happens if they *don't* optimize? I know they're supposed to import and use the webpack config that the LDK provides. But like what if they set their webpack config to not minify (or disable whatever other optimization)? I do believe I've seen something like this before - the dev was doing something different/special when in development mode around optimization. Still, we probably don't care about `development` mode giving a false negative or something.

Reply:
yeah thats another good point. We could have both hooks in place as an ultimate fail safe but assume (probably dangerous) that users will optimize